SDWebImage/NSData+ImageContentType.m

```diff
@@ -21,6 +21,7 @@ + (SDImageFormat)sd_imageFormatForImageData:(nullable NSData *)data {
         return SDImageFormatUndefined;
     }
     
+    // File signatures table: http://www.garykessler.net/library/file_sigs.html
     uint8_t c;
     [data getBytes:&c length:1];
     switch (c) {
```

Comment:
Actually. This kinid of file type detect is totally wrong. For example, JPEG format should check `FF 08` and PNG should check `89 50 4E 47`. Only check the first byte is really tricky and may leak wrong image format detect if some image format have the same byte as others.

Reply:
I suggest to change this and check the image format one by one with their file sig. But this may be a carefully changes. Any idea ?

Reply:
I've also noticed this fact (that we don't do textbook checks for the image format). But I'm concerned doing it, as this might break for some people. I'm not sure all the PNG files have the whole sequence `89 50 4E 47` or JPEG `FF 08`.
I would like to improve it, but I think it's better if we leave it alone.

Reply:
@bpoplauschi For file format, it should have this file signatures(or sometimes called magic number). I just means that the first byte test is a little tricky. Maybe we need to check that specify bytes match the actual file signatures for detecting format. But currently it works...So, it's a question.